### PR TITLE
ansible: Fix for Linuxbrew

### PR DESCRIPTION
**Pull request by hartzell**
### All Submissions:

- [x] Have you followed the guidelines in our [Contributing](https://github.com/Homebrew/homebrew/blob/master/.github/CONTRIBUTING.md) document?
- [x] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/Homebrew/homebrew/pulls) for the same update/change?

---

The Babel package died looking for pytz.  If I move the pytz
resource ahead of Babel's then it works.

### CODE REVIEW

**Comment by hartzell** [comment]
I'd like to keep plugging away at this but could use some guidance.

I've been trying to get things going on real live linux boxes, but that might be too confusing (I have no control of what's installed where when...).

I'm working on getting the linuxbrew docker images/containers up and running.

**Comment by sjackman** [comment]
Thanks for your contribution, George. I've added the constant `OS::NAME` in ce0f667 just now to make your patch easier. Its value is either `darwin` on Mac OS or `linux` on Linux.

File: Library/Formula/ansible.rb [hunk 1]
```diff
@@ -22,6 +22,7 @@ class Ansible < Formula
   depends_on :python if MacOS.version <= :snow_leopard
   depends_on "libyaml"
   depends_on "openssl"
+  depends_on "libffi" if OS.linux?
```
Comment (sjackman) [inline Library/Formula/ansible.rb]: Please change this line to
```ruby
depends_on "libffi" unless OS.mac?
```

Comment (sjackman) [inline Library/Formula/ansible.rb]: Also please report the error message that you see without this patch. `brew gist-logs ansible` is helpful.

File: Library/Formula/ansible.rb [hunk 2]
```diff
@@ -331,11 +337,6 @@ class Ansible < Formula
     sha256 "b00a70ba72b068adb6aaca19b14ed4bde9c950ef4f24234d095f9f9985073179"
   end
 
-  resource "pytz" do
-    url "https://pypi.python.org/packages/source/p/pytz/pytz-2015.7.tar.bz2"
-    sha256 "fbd26746772c24cb93c8b97cbdad5cb9e46c86bbdb1b9d8a743ee00e2fb1fc5d"
-  end
-
```
Comment (sjackman) [inline Library/Formula/ansible.rb]: Please remove this chunk of the patch if it has no purpose.

**Comment by sjackman** [comment]
Are you still working on this patch, George?

File: Library/Formula/ansible.rb (hunk 2 above)
Comment (hartzell) [inline Library/Formula/ansible.rb]: I'm not sure which hunk you'd like me to remove.  The patch moves pytz to a different location in the file, the first hunk inserts it into the new location and the second hunk deletes it from the old.

File: Library/Formula/ansible.rb (hunk 1 above)
Comment (hartzell) [inline Library/Formula/ansible.rb]: https://gist.github.com/efdb86973be7483a4592

Comment (hartzell) [inline Library/Formula/ansible.rb]: Digging a bit more.  I've made the change to use `unless OS.mac?`.  I'm still having trouble.

I think that the problem is that the *libffi* formula does not install *ffi.h` into the include directory, so building cffi fails.

*Should* the *libffi* formula copy/link/... *ffi.h*?

Comment (sjackman) [inline Library/Formula/ansible.rb]: The `ffi` formula installs `ffi.h` in a subdirectory, which seems to be the default behaviour of `libffi`.
```
$ brew list libffi |grep ffi.h
/gsc/btl/linuxbrew/Cellar/libffi/3.0.13/lib/libffi-3.0.13/include/ffi.h
$ ls `brew --prefix libffi`/lib/libffi-3.0.13/include
ffi.h  ffitarget.h
```

File: Library/Formula/ansible.rb (hunk 2 above)
Comment (sjackman) [inline Library/Formula/ansible.rb]: Ah, I see. I didn't realize that the Python resources are installed in the order that they're listed in this file by magics later on. Sorry for my confusing comment.

File: Library/Formula/ansible.rb (hunk 1 above)
Comment (sjackman) [inline Library/Formula/ansible.rb]: > I've made the change to use unless OS.mac?

I don't see this change yet. You can update your PR with `brew push --force`

**Comment by sjackman** [comment]
Please squash the two commits to a single commit, and use set the first line of the log message to
`ansible: Fix for Linuxbrew`
The rest of the log message can be whatever you want.
To squash the two commits run
```sh
git checkout ansible-on-linux
brew rebase -i HEAD~2
git push --force hartzell ansible-on-linux
```
I'll merge this PR after the squash.

**Comment by hartzell** [comment]
Squashed and pushed.

By the way, the squash/rebase instructions were *very* helpful.  I probably would have screwed around a while and possibly blown it.  Three lines from you made the commit happen easily.  (Although the second line is `git`, not `brew`...).

**Comment by sjackman** [comment]
> By the way, the squash/rebase instructions were very helpful.

Glad to hear it! I'll add those instructions to the `CONTRIBUTING.md`.

> (Although the second line is git, not brew...).

Fixed above, for posterity. Thanks.

**Comment by sjackman** [comment]
Merged. Thanks for your contribution, George, and for sticking with it!

**Pull request closed without merging**